inst/htmlwidgets/datatables.js

```diff
@@ -1402,7 +1402,8 @@ HTMLWidgets.widget({
           console.log('The search keyword for column ' + i + ' is undefined')
           return;
         }
-        $(td).find('input').first().val(v).trigger('input');
+        // Change event is needed to update numeric slider values.
+        $(td).find('input').first().val(v).trigger('input').trigger('change');
```

Comment:
If we trigger the `change` event, it seems `searchColumn()` on the next line is no longer necessary. If that sounds correct to you, would you help test it?

Reply:
Yep it looks like it will get triggered by the events anyway. Quickly testing removing it, the numeric filter in the example here seems to keep working fine. I'm happy to have a more thorough look at the other filter types tomorrow.

Reply:
I tested removing it with all 3 filter types now, using the app below. As far as I can tell, everything keeps working as intended. I also logged the calls to `searchColumn()` and they happen as intended. I pushed the change.

``` r
library(shiny)
library(DT)

tbl <- data.frame(
  num1 = seq(-2, 2) / 10,
  fct1 = factor(c("A", rep("B", 3), "C")),
  chr = c("foo", "bar", "baz", "baz", "baz")
)

ui <- fluidPage(
  DTOutput("table"),
  actionButton("update_search", "Update search"),
  actionButton("clear_search", "Clear search"),
)

server <- function(input, output, session) {
  output$table <- renderDT(datatable(tbl, filter = "top", rownames = FALSE))
  
  proxy <- dataTableProxy("table") 
  observeEvent(input$update_search, {
    updateSearch(proxy, list(columns = c("-0.1 ... 0.1", '["B"]', 'bar')))
  })
  observeEvent(input$clear_search, clearSearch(proxy))
}

shinyApp(ui, server)
```

Reply:
Although now I do notice that sometimes when updating multiple search strings at a time, the search based on the numeric column happens first, and it takes a few moments for the other to kick in. That seems undesirable. Probably related to always `.draw()`ing when searching.

Reply:
Okay the delay is because of the character search field input throttling and only happens because when testing I'm flicking quickly between "update" and "clear". It's made apparent because the numeric search triggers the `.draw()` early. Not sure why the factor filter wouldn't do the same even without this patch, though.
https://github.com/rstudio/DT/blob/487dc08fe757fe0be68931c8b88395f11a9c155a/inst/htmlwidgets/datatables.js#L533-L535

Reply:
Oh I see it's because the removed `searchColumn()` was bypassing the throttling on the character search.